app/controllers/katello/api/v2/packages_controller.rb

```diff
@@ -19,6 +21,8 @@ def auto_complete_arch
       rpms = Rpm.in_repositories(@repositories)
       col = "#{Rpm.table_name}.arch"
       rpms = rpms.where("#{col} ILIKE ?", "%#{params[:term]}%").select(col).group(col).order(col).limit(page_size)
+      rpms = rpms.modular if Foreman::Cast.to_bool(params[:modular_only])
+      rpms = rpms.non_modular if Foreman::Cast.to_bool(params[:non_modular_only])
```

Comment:
We could also have params like modular_only = false , non_modular_only =false?

Reply:
modular, non modular and both are the 3 options possible here. If I just included one flag for all 3 options it could get more confusing. I'll have to add logic that says
```
if param not set:
  dont filter by modular rpms
else:
   if its true:
      filter by modular rpms
  else:
      filter by non-modular 
```
More over even from the users perspective I think it is clearer for the user to say "modular_only=true" and false implies there won't be any filtering. We do this kinda thing in other controllers as well. 
https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/products_controller.rb#L38  for example.


Setting  modular_only and non_modular